app/order-history/page.tsx

```diff
@@ -0,0 +1,381 @@
+              className="ml-2 text-gray-500 hover:text-gray-700"
+            >
+              <X className="h-4 w-4" />
+            </button>
+          </div>
+        ))}
+
+        {logisticsFilter && (
+          <div
+            className="flex items-center bg-gray-200 px-3 py-1 rounded-full text-gray-700"
+          >
+            {logisticsFilter}
+            <button
+              onClick={() => setLogisticsFilter(null)}
+              className="ml-2 text-gray-500 hover:text-gray-700"
+            >
+              <X className="h-4 w-4" />
+            </button>
+          </div>
+        )}
+
+        {statusFilter && (
+          <div
+            className="flex items-center bg-gray-200 px-3 py-1 rounded-full text-gray-700"
+          >
+            {statusFilter}
+            <button
+              onClick={() => setStatusFilter(null)}
+              className="ml-2 text-gray-500 hover:text-gray-700"
+            >
+              <X className="h-4 w-4" />
+            </button>
+          </div>
+        )}
+
+        {paymentStatusFilter && (
+          <div
+            className="flex items-center bg-gray-200 px-3 py-1 rounded-full text-gray-700"
+          >
+            {paymentStatusFilter}
+            <button
+              onClick={() => setPaymentStatusFilter(null)}
+              className="ml-2 text-gray-500 hover:text-gray-700"
+            >
+              <X className="h-4 w-4" />
+            </button>
+          </div>
+        )}
+
+        {/* Clear all filters Button */}
+        <button
+          onClick={clearFilters}
+          disabled={!filter && serviceFilter.length === 0 && !logisticsFilter && !statusFilter && !paymentStatusFilter}
+          className={`ml-2 px-4 py-2 rounded-lg transition-all ${filter || serviceFilter.length > 0 || logisticsFilter || statusFilter || paymentStatusFilter ? "bg-[#9CC0D9] text-white hover:bg-blue-700" : "bg-gray-300 text-gray-500 cursor-not-allowed"}`}
+        >
+          Clear All
+        </button>
+      </div>
+
+      <div className="border border-gray-300 shadow-lg rounded-2xl overflow-hidden mt-6 p-4">
+
+      {/* Table of Orders */}
+      <Table className="min-w-full table-auto rounded-2xl overflow-hidden">
+
+        {/* Table Headers */}
+        <thead
+          className="text-left text-gray-700 uppercase tracking-wide text-sm font-semibold rounded-t-2xl"
+          style={{ backgroundColor: '#DBEAFF' }}
+        >
+          <tr className="border-b border-gray-200">
+            {tableHeaders.map((header, index) => (
+              <th key={index} className="py-2 px-4">{header}</th>
+            ))}
+          </tr>
+        </thead>
+
+        {/* Table Body */}
+        <tbody className="text-gray-600">
+          {filteredOrders.length > 0 ? (
+            filteredOrders.map((order, index) => ( //Map for filters
+              <tr key={index} className="border-b border-gray-200">
```

Comment:
again dont use index for the key, think of something unique to identify each created element